app/client/pages/CreateListingPage.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Card from 'react-bootstrap/Card';
+import logo from '../../assets/images/logo/black_transparent.svg';
+import CreateListing from '../components/CreateListing.jsx';
+import Redirect from '../components/routing/Redirect.jsx';
+import Splash from '../components/layouts/Splash.jsx';
+import { withSession } from '../components/Remote.jsx';
+import selectors from './CreateListingPage.module.scss';
+
+/** Login */
+class CreateListingPage extends React.Component {
+  /** @param {Object} props */
+  constructor(props) {
+    super(props);
+    this.handleAddListing = this.handleAddListing.bind(this);
+  }
+
+  /** @param {Object} credentials */
+  handleAddListing(credentials) {
+    this.props.postRemote('/sessions', {
+      body: credentials,
+      success: (response) => this.props.updateSession(response.body || {})
+    });
+  }
```

Comment:
```suggestion
  /** @param {object} listing */
  handleSubmit(listing) {
    this.props.fetch('/listings', {
      method: 'POST',
      body: listing,
      success: (response) => {
        // update state so that a redirect is rendered
        // this.setState({ submit: true });
      },
      error: (response) => {
        // update `this.state.errors`
      }
    });
  }
```

This sprawls out a bit to the rest of the component:
- note the rename - make sure to update the usage of this method in `render` as well as the binding in `constructor`
- have to specify `this.state.errors` as an array in constructor and forward it to the form as a prop
- have to add something some sort of boolean flag like `this.state.submit` and edit the render function so that a `<Redirect back/>` component is rendered when the success function set it.